src/stringify/stringifyPair.ts

```diff
@@ -127,7 +127,7 @@ export function stringifyPair(
       ws += `\n${indentComment(cs, ctx.indent)}`
     }
     if (valueStr === '' && !ctx.inFlow) {
-      if (ws === '\n') ws = '\n\n'
+      if (vsb && !vcb && valueComment) ws = '\n\n'
```

Comment:
Let's extend the previous conditional here.
```suggestion
      if (ws === '\n' && valueComment) ws = '\n\n'
```

Reply:
Sounds good.